contrib/flake.nix

```diff
@@ -7,13 +7,50 @@
   };
 
   outputs = { self, nixpkgs, flake-utils }:
+    let
+      inherit (builtins)
+        elemAt
+        foldl'
+        mapAttrs
+        match
+        readFile
+        ;
+      inherit (nixpkgs.lib)
+        const
+        flip
+        pipe
+        remove
+        splitString
+        toLower
+        ;
+    in
     {
       overlay = final: prev: {
 
-        neovim = final.neovim-unwrapped.overrideAttrs (oa: rec {
+        neovim = (final.neovim-unwrapped.override {
```

Comment:
@dundargoc feel free to flat out say no but I wonder if we could make the deps.txt file more easily consumable by packagers, like json format , seems like that could work with native cmake https://stackoverflow.com/questions/46703437/cmakelists-parse-file.

Reply:
We'd need to bump minimum cmake version to 3.19. Not a flat out no, I just want to emphasize this is a big version requirement spike and that we'd need to convince people who will have opinions about this why this is needed.

Reply:
Any roadblock to do that ? even ubuntu has 3.24 https://packages.ubuntu.com/kinetic/cmake. I will merge this without it since cmake upgrade seems like a big deal.

Reply:
Not the version of Ubuntu we target, unfortunately. (Nor Debian, and we don't want to make it even harder for jamessan to finally package Neovim again.)

Reply:
@justinmk @jamessan

Reply:
Here's the current support matrix:
### Debian
* Buster (oldstable, LTS until June 2024): 3.13 (backports: 3.18)
* Bullseye (stable, LTS until June 2026): 3.18 (backports: 3.25)
* Bookworm (new stable): 3.25

### Ubuntu
* 20.04 (LTS until 2025): 3.16
* 22.04 (LTS until 2027): 3.22
* 22.10: 3.24
* 23.04: 3.25

So we'd have to drop support for Debian 10 and Ubuntu 20.04, and require Debian 11 to tap backports.

Reply:
Let's make a tracking issue for now. Definitely something to consider in the future.

Reply:
That is a good idea. Makes it easier to refer to. @dundargoc If you have the time, you could open it with a list of features such a bump would buy us (which can be extended later)?